storage/rocksdb/rdb_datadic.cc

```diff
@@ -6033,6 +6038,40 @@ bool Rdb_dict_manager::update_max_index_id(rocksdb::WriteBatch *const batch,
   return false;
 }
 
+bool Rdb_dict_manager::get_server_version(uint *const ser_version) const {
+  std::string value;
+
+  const rocksdb::Status status = get_value(m_key_slice_server_version, &value);
+  if (status.ok()) {
+    const uchar *const val = (const uchar *)value.c_str();
+    const uint16_t version = rdb_netbuf_to_uint16(val);
+    if (version == Rdb_key_def::SERVER_VERSION_VERSION) {
+      *ser_version = rdb_netbuf_to_uint32(val + Rdb_key_def::VERSION_SIZE);
+      return false;
+    }
+  }
+
+  return true;
+}
+
+bool Rdb_dict_manager::set_server_version() const {
+  const std::unique_ptr<rocksdb::WriteBatch> wb = begin();
+  rocksdb::WriteBatch *const batch = wb.get();
```

Comment:
Why do we mirror a smart pointer in a raw pointer? IMHO removing the raw one, renaming `wb` to `batch`, and using `batch.get()` inline as needed would result in as clean or cleaner code